server/src/testFixtures/java/org/elasticsearch/test/IntegTestCase.java

```diff
@@ -275,7 +277,7 @@
 @UseJdbc
 @UseHashJoins
 @UseRandomizedSchema
-@UseRandomizedOptimizerRules(alwaysKeep = {RemoveRedundantFetchOrEval.class, MergeFilterAndCollect.class})
+@UseRandomizedOptimizerRules(alwaysKeep = {RewriteJoinPlan.class, EliminateCrossJoin.class, RemoveRedundantFetchOrEval.class, MergeFilterAndCollect.class})
```

Comment:
Shouldn't the tests also pass without these rules?

Reply:
Another alternative could be to operate on the NestedLoopJoin and drop the whole JoinPlan concept. The JoinPlan prevents other rules from interacting with the Joins before the reordering is done, but this can also be accomplished in other ways.

Reply:
Could we apply `RewriteJoinPlan` in a different way? Users shouldn't have the option to disable a rule that's mandatory.

(Similar to how `RewriteInsertFromSubQueryToInsertFromValues` is also mandatory)

Reply:
Otherwise I think I'd prefer this:

> . We could consider falling back in the build method of JoinPlan to Nested loop join to make sure we always have a working version.

Reply:
We could add a property to the rule which indicates that it can be disabled or make a set of static mandatory rules.

Reply:
I would prefer to add a fallback from join-plan to nested loop-join for this pr and then do a follow-up to improve the situation for mandatory rules.